rskj-core/src/main/java/org/ethereum/net/NodeManager.java

```diff
@@ -119,24 +108,17 @@ public NodeStatistics getNodeStatistics(Node n) {
         return handlers;
     }
 
-    public Node getHomeNode() {
-        return this.homeNode;
-    }
-
-    public Boolean inited() {
-        return inited;
-    }
-
     private void purgeNodeHandlers() {
         if (nodeHandlerMap.size() > NODES_TRIM_THRESHOLD) {
-            List<NodeHandler> sorted = new ArrayList<>(nodeHandlerMap.values());
-            Collections.sort(sorted, (o1, o2) -> Integer.compare(o1.getNodeStatistics().getReputation(), o2.getNodeStatistics().getReputation()));
-            for (NodeHandler handler : sorted) {
-                nodeHandlerMap.remove(handler.getNode().getAddressAsString());
-                if (nodeHandlerMap.size() <= MAX_NODES) {
-                    break;
-                }
-            }
+            //I create a stream
+            List<NodeHandler> toRemove = nodeHandlerMap.values().stream()
+                    //sort by reputation
+                    .sorted(Comparator.comparingInt(o -> o.getNodeStatistics().getReputation()))
+                    //and just keep the ones that exceeds the MAX_NODES
+                    .limit(nodeHandlerMap.size() - MAX_NODES)
```

Comment:
If you have 3001 nodes you want to keep 1001 or 2000?

Reply:
:+1: